designs/data-lake.md

```diff
@@ -0,0 +1,114 @@
+  containers:
+    - name: Transformed
+- name: Gold
+  containers:
+    - name: Curated
+```
+
+The objects under zones represent the data lakes to deploy (an Azure Data Lake Storage gen2 Filesystem), and the containers represent the storage container(s) within each filesystem.
+
+#### Infrastructure
+
+The new infrastructure to deploy will live in the `infrastructure/transform` Terraform module and will comprise of the following:
+
+- Storage Account with Heirarchical Namespaces enabled (Azure Data Lake Storage Gen2)
+- ADLS Gen2 Filesystem(s) (for each configured zone)
+- Storage Container(s) (a `azurerm_storage_data_lake_gen2_path` for each configured container)
+- ADLS private networking (private endpoint, dns zone and private link)
+- ADLS role assignments for Azure Data Factory and Databricks
+- Databricks ADLS filesystem mounts for each zone
+- Service principal for Databricks identity access to ADLS
+- Databricks secrets for SP credentials
+- ADF linked service for ADLS
+
+As mentioned, if the `datalake` configuration block is not present, none of this infrastructure will be deployed by utilising a dynamic flag.
+
+
+## Considerations
+
+- What do we do with the SQL DB now or longer term? Do we deprecate it in favour of the data lake?
+
+Given that the current "gold" layer in FlowEHR is the SQL DB, we should consider whether it would be superseded by ADLS. This will greatly simplify the architecture of FlowEHR; however the SQL Server currently also serves as the query layer for applications and data scientists / ML models.
+
+We would need to provide another mechanism for querying and analysing data in the data lake, which could include:
+- Through the existing [Databricks workspace](https://learn.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-use-databricks-spark)
+    - this would be the most seamless option but would require users and apps/models to have access to Databricks, which might not be viable (i.e. users might be accessing from a TRE)
+    - [Delta Lake](https://learn.microsoft.com/en-us/azure/databricks/delta/) could be enabled to further optimise for analytics
+- [Azure Data Explorer](https://learn.microsoft.com/en-us/azure/data-explorer/data-lake-query-data)
+    - Multiple SDKs including Python that can be used to query data but another service to add into the mix
+- [Azure Machine Learning Datastore](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-datastore?view=azureml-api-2&tabs=sdk-identity-based-access,sdk-adls-identity-access,sdk-azfiles-accountkey,sdk-adlsgen1-identity-access)
+    - Useful for ML workloads but not for general querying
+
+An alternative might be that the SQL DB is shifted to the right, and is considered just a feature store layer on top of gold for the time being. This might eventually be replaced by a tailor-made feature store like [AML Feature Store](https://learn.microsoft.com/en-us/azure/machine-learning/concept-what-is-managed-feature-store?view=azureml-api-2).
```

Comment:
I think this is just another option above, and would be the easier way forward initially. Then in future a move to a tailor made feature store.

Reply:
Yeah I mention it in the considerations but think you're right maybe worth calling out as its own section - feel free to add!